### PR TITLE
Company Profile and Review forms (and some bug fixes)

### PR DESCRIPTION
**Pull request by jhigginbotham64**
Got the form and associated Method to work. Probably need to see if there's a way to test the finer aspects of schema validation, but the important bits seem to work: it catches duplicates, rejects a user that isn't logged in, and (most importantly) indeed updates the underlying collection.
Things to add: better error reporting, better looks, proper integration with the site's structure
Things to remove: schmucky testing code, which will be left in until the other necessary changes have been made or I find a better way to test the code
IMPORTANT THINGS ABOUT THIS NEW CODE
I found some bugs in methods.js (and denormalization.js by extension, though I commented it out instead of trying to fix it) and the server's main.js, as well as added a number of important packages.
Most important among these new packages is gadicc:blaze-react-component. This allowed me to wrap a Blaze package template (from aldeed:autoform) into a React component, which in turn allowed me to...implement the form, period.
**This could be a pretty landmark find: in theory, it allows us to use any old Blaze package from inside our React code.** Plus, Blaze is just cool, as I've learned. But you have no idea, the whole Blaze/React incompatibility problem was _killing Krit and I's workflow_, so it's a major relief to have found something that fixes it, even if the fix is temporary.
There's still a lot to learn about it. For instance, I have no idea if and how React's reactivity integrate with Blaze's and vice versa: the code I have is pretty much just running Blaze code from within React, I have no idea if more complex interactions are possible. But then again, they may not even be necessary. Who knows. In any case, autoforms is a cool package and I like being able to use it.
Other interesting tidbits include some of the testing packages I found, such as csauer:accounts-phony, faker, and dburles:factory. Also interesting was getting to see for the first time how aldeed:collection2 and simpl-schema interact with aldeed:autoform.
All of this should be visible in the code (and hopefully in the comments).
Please don't merge immediately, even if for some strange reason GitHub lets you, without really reviewing this code. It needs work.

### CODE REVIEW

File: imports/api/data/methods.js [hunk 1]
```diff
@@ -1,55 +1,59 @@
 import { Reviews } from "./reviews.js";
 import { Companies } from "./companies.js";
-import "./denormalizers.js"
```
Comment (joshuatshaffer) [inline imports/api/data/methods.js]: This is importing the wrong file. The real file is called denormalization.js. This never caused an error for me because JavaScript does not give compile errors like any sensible language would.

Comment (jhigginbotham64) [reply imports/api/data/methods.js]: Fixed.

**Comment by jhigginbotham64** [comment]
Hey guys, please review this pull request. I fixed the major issues and reached the milestones I had set in the initial request, as well as removed all the testing code and did some general cleanup, so it should be ready for actual review now.

**Comment by jhigginbotham64** [comment]
I will give my branches better names in the future.
